From 39f8516bfabab31fb062fa78554e6083bfa98d63 Mon Sep 17 00:00:00 2001
From: Ray Donnelly <mingw.android@gmail.com>
Date: Wed, 31 Oct 2018 12:28:22 +0000
Subject: [PATCH 4/4] Export LIEF::Object and LIEF::MachO::DataCodeEntry
 classes

Otherwise the shared library does not contain these symbols.
---
 include/LIEF/MachO/DataCodeEntry.hpp | 2 +-
 include/LIEF/Object.hpp              | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/include/LIEF/MachO/DataCodeEntry.hpp b/include/LIEF/MachO/DataCodeEntry.hpp
index bff5d77..37d7ec6 100644
--- a/include/LIEF/MachO/DataCodeEntry.hpp
+++ b/include/LIEF/MachO/DataCodeEntry.hpp
@@ -29,7 +29,7 @@ namespace LIEF {
 namespace MachO {
 
 //! Interface of an entry in DataInCode
-class DataCodeEntry : public LIEF::Object {
+class LIEF_API DataCodeEntry : public LIEF::Object {
   public:
   enum class TYPES {
     UNKNOWN           = 0,
diff --git a/include/LIEF/Object.hpp b/include/LIEF/Object.hpp
index def2fb0..4f727fb 100644
--- a/include/LIEF/Object.hpp
+++ b/include/LIEF/Object.hpp
@@ -30,7 +30,7 @@ using add_const_t = typename std::add_const<T>::type;
 
 namespace LIEF {
 
-class Object {
+class LIEF_API Object {
   public:
   template<class T>
   using output_t = add_pointer_t<decay_t<T>>;
-- 
2.19.1

